src/ansys/fluent/core/solver/flobject.py

```diff
@@ -243,6 +244,33 @@ def set_state(self, state: StateT = None, **kwargs):
         else:
             return self.flproxy.set_var(self.path, self.to_scheme_keys(state))
 
+    def find_child(self, identifier="*"):
```

Comment:
Shouldn't this be called `find_children` if it is returning multiple children?

Should we rename `identifier` as `pattern` if it is a wild-card pattern?

Does this recurse to all children and return all children matching that pattern? If so, do we need a way to disable recursion?

Reply:
find_children sounds good, we can update that and the identifier to pattern as well. Thank you for the suggestion.
Yes, it currently recurse to all children and return all children matching that pattern, @seanpearsonuk, do we need this behavior or stop the recursion?

Reply:
Should we return paths or the objects themselves? What can users do with the path other than get the object with it?

Reply:
Please note that the current implementation is based on static information. It doesn't resolve the named objects nor filter the inactive children. This is to help users to write scripts before reading the case or setting up the model. Maybe a more appropriate name would be `search_child_paths` for this function.

Reply:
So, this is for use in the editor and not in actual scripts?

Reply:
just to clarify on the above discussion:
>>> solver.setup.boundary_conditions.velocity_inlet.find_child("vmag*")
['phase/vmag', 'vmag']

This is the current behavior.

Reply:
All the suggestions provided by @h-krishnan have been implemented. The ListObjects are covered by this as well now. Can this go in as the first version of find_child() and later we do any modifications as and when required? What do you think @h-krishnan , @seanpearsonuk ?

Reply:
Hi @prmukherj, just recalling the `find_children` suggestion from @h-krishnan.

Reply:
Yes @seanpearsonuk, this sounds more appropriate and has been renamed accordingly. Thank you for pointing it out.